app/models/page_content_register.rb

```diff
@@ -26,6 +31,34 @@ def new_blocks_attributes
 
   private
 
+  def srcset_blocks(blk_attrs: [])
+    blk_attrs
+      .select { |b| b[:identifier] =~ home_page_tile_identifer }
+      .map do |b|
+      {
+        identifier: b[:identifier].sub('image', 'srcset'),
+        content: srcset_content(original_url: b[:content]).join(', ')
+      }
+    end
+  end
+
+  def available_styles
+    Comfy::Cms::File.new.file.styles
+      .reject { |k,_| [:cms_thumb, :cms_medium].include? k }
+      .map do |style, object|
+      {
+        style: style,
+        width: object.geometry.split('x').first
+      }
+    end
+  end
```

Comment:
The next best thing is probably to stick this in the libs directory or make it a model concern. Wdyt?